File: Sources/custom_adapter/TopOnAdapter/Objc/6.5.x/Interstitial/AlxTopOnInterstitialEvent.m
//
//  AlxTopOnInterstitialEvent.m
//  AlxAdsOCDemo
//

#import "AlxTopOnInterstitialEvent.h"
#import "AlxTopOnBaseManager.h"
#import "AlxTopOnBiddingRequestManager.h"

static NSString *const TAG = @"AlxTopOnInterstitialEvent";

@implementation AlxTopOnInterstitialEvent

- (void)interstitialAdLoad:(AlxInterstitialAd *)ad {
    NSLog(@"%@: interstitialAdLoad", TAG);
    if (self.isC2SBiding) {
        [AlxTopOnBiddingRequestManager disposeLoadSuccessWithPrice:[ad getPrice]
                                                            unitID:self.serverInfo[[AlxTopOnBaseManager unitID]]];
        self.isC2SBiding = NO;
    } else {
        [self trackInterstitialAdLoaded:ad adExtra:nil];
    }
}

- (void)interstitialAdFailToLoad:(AlxInterstitialAd *)ad didFailWithError:(NSError *)error {
    NSLog(@"%@: interstitialAdFailToLoad : %@", TAG, error.localizedDescription);
    if (self.isC2SBiding) {
        [AlxTopOnBiddingRequestManager disposeLoadFailWithError:error
                                                         unitID:self.serverInfo[[AlxTopOnBaseManager unitID]]];
    } else {
        [self trackInterstitialAdLoadFailed:error];
    }
}

- (void)interstitialAdImpression:(AlxInterstitialAd *)ad {
    NSLog(@"%@: interstitialAdImpression", TAG);
    [self trackInterstitialAdShow];
}

- (void)interstitialAdClick:(AlxInterstitialAd *)ad {
    NSLog(@"%@: interstitialAdClick", TAG);
    [self trackInterstitialAdClick];
}

- (void)interstitialAdClose:(AlxInterstitialAd *)ad {
    NSLog(@"%@: interstitialAdClose", TAG);
    [self trackInterstitialAdClose:nil];
}

- (void)interstitialAdRenderFail:(AlxInterstitialAd *)ad didFailWithError:(NSError *)error {
    NSLog(@"%@: interstitialAdRenderFail", TAG);
}

- (void)interstitialAdVideoStart:(AlxInterstitialAd *)ad {
    NSLog(@"%@: interstitialAdVideoStart", TAG);
    [self trackInterstitialAdVideoStart];
}

- (void)interstitialAdVideoEnd:(AlxInterstitialAd *)ad {
    NSLog(@"%@: interstitialAdVideoEnd", TAG);
    [self trackInterstitialAdVideoEnd];
}

- (NSString *)networkUnitId {
    return self.serverInfo[[AlxTopOnBaseManager unitID]] ?: @"";
}

- (void)dealloc {
    NSLog(@"%@: dealloc", TAG);
}

@end

File: Sources/custom_adapter/TopOnAdapter/Swift/6.4.x/AlxTopOnInterstitialEvent.swift
//
//  AlxTopOnInterstitialEvent.swift
//


import Foundation
import AnyThinkSDK
import AnyThinkInterstitial
import AlxAds

@objc(AlxTopOnInterstitialEvent)
public class AlxTopOnInterstitialEvent: ATInterstitialCustomEvent,AlxInterstitialAdDelegate {
    
    private static let TAG = "AlxTopOnInterstitialEvent"
    
    public override init(info serverInfo: [AnyHashable : Any], localInfo: [AnyHashable : Any]) {
        super.init(info: serverInfo, localInfo: localInfo)
    }
    
    public func interstitialAdLoad(_ ad: AlxInterstitialAd) {
        NSLog("%@: interstitialAdLoad",AlxTopOnInterstitialEvent.TAG)
        if self.isC2SBiding {
            AlxTopOnBiddingRequestManager.disposeLoadSuccess(price: ad.getPrice(), unitID: self.serverInfo[AlxTopOnBaseManager.unitID] as? String)
            self.isC2SBiding = false
        }else{
            self.trackInterstitialAdLoaded(ad, adExtra: nil)
        }
    }
    
    public func interstitialAdFailToLoad(_ ad: AlxInterstitialAd, didFailWithError error: Error) {
        NSLog("%@: interstitialAdFailToLoad : %@",AlxTopOnInterstitialEvent.TAG,error.localizedDescription)
        if self.isC2SBiding {
            AlxTopOnBiddingRequestManager.disposeLoadFail(error: error, unitID: self.serverInfo[AlxTopOnBaseManager.unitID] as? String)
        }else{
            self.trackInterstitialAdLoadFailed(error)
        }
    }
    
    public func interstitialAdImpression(_ ad: AlxInterstitialAd) {
        NSLog("%@: interstitialAdImpression",AlxTopOnInterstitialEvent.TAG)
        self.trackInterstitialAdShow()
    }
    
    public func interstitialAdClick(_ ad: AlxInterstitialAd) {
        NSLog("%@: interstitialAdClick",AlxTopOnInterstitialEvent.TAG)
        self.trackInterstitialAdClick()
    }
    
    public func interstitialAdClose(_ ad: AlxInterstitialAd) {
        NSLog("%@: interstitialAdClose",AlxTopOnInterstitialEvent.TAG)
        self.trackInterstitialAdClose(nil)
    }
    
    public func interstitialAdRenderFail(_ ad: AlxInterstitialAd, didFailWithError error: Error) {
        NSLog("%@: interstitialAdRenderFail",AlxTopOnInterstitialEvent.TAG)
    }
    
    public func interstitialAdVideoStart(_ ad: AlxInterstitialAd) {
        NSLog("%@: interstitialAdVideoStart",AlxTopOnInterstitialEvent.TAG)
        self.trackInterstitialAdVideoStart()
    }
    
    public func interstitialAdVideoEnd(_ ad: AlxInterstitialAd) {
        NSLog("%@: interstitialAdVideoEnd",AlxTopOnInterstitialEvent.TAG)
        self.trackInterstitialAdVideoEnd()
    }
    
    public override var networkUnitId: String {
        get {
            return self.serverInfo[AlxTopOnBaseManager.unitID] as? String ?? ""
        }
        set {
        }
    }
    
    deinit {
        NSLog("%@: deinit",AlxTopOnInterstitialEvent.TAG)
    }
    
}
